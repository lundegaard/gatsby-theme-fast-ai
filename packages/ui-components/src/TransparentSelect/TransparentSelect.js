import React from 'react';
import { Select as RebassSelect } from '@rebass/forms';

const TransparentSelect = props => (
	<RebassSelect
		sx={{
			color: 'inherit',
			border: 'none',
			fontSize: [2, 2, 2, 4],
			WebkitTapHighlightColor: 'transparent',
			// FF
			'&:invalid': {
				boxShadow: 'none',
			},
			'&:focus': {
				outline: 0,
			},
			width: '100%',
		}}
		px={0}
		{...props}
	/>
);

export default TransparentSelect;
