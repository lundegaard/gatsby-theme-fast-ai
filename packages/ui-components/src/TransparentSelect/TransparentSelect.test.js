import React from 'react';
import { shallow } from 'enzyme';

import TransparentSelect from './TransparentSelect';

describe('<TransparentSelect />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<TransparentSelect {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
