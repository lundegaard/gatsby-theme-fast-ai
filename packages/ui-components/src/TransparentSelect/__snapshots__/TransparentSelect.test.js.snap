// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransparentSelect /> matches snapshot when shallowly rendered 1`] = `
<ForwardRef
  px={0}
  sx={
    Object {
      "&:focus": Object {
        "outline": 0,
      },
      "&:invalid": Object {
        "boxShadow": "none",
      },
      "WebkitTapHighlightColor": "transparent",
      "border": "none",
      "color": "inherit",
      "fontSize": Array [
        2,
        2,
        2,
        4,
      ],
      "width": "100%",
    }
  }
/>
`;
